common/scripts/changelog/utils.mjs

```diff
@@ -1,11 +1,10 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-const NEW_CHANGE_FILE_REGEXP = /\s*change\/(.*\.json)\s*/;
-
 export function parseNewChangeFiles(stdout) {
     const lines = stdout.split('\n');
-    const matches = lines.map(line => line.match(NEW_CHANGE_FILE_REGEXP)).filter(match => !!match);
-    // Extract the first capture group.
-    return matches.map(match => match[1]);
+    // Filter out the lines that are not changefiles
+    const matches = lines.filter(line => line.startsWith('A'));
```

Comment:
not in this case. `git diff --name-status` returns files for `changedStableFiles` and `changedBetaFiles` in
```
A       change-beta/@azure-communication-react-3ba4c269-8d22-48da-b905-091fdbaf2f98.json
A       change-beta/@azure-communication-react-c7a5e0c3-2c59-4093-8334-9e705a8feb89.json
```
format

To check:
- pull this branch
- add console logs for `changedStableFiles` and `changedBetaFiles`
- run `git diff --diff-filter=A --name-status 395e3e5d2caa04be66840849a936f13de732ee53..75c78e74eb3707dda151721f2debe85e51bf1e42 -- <change-logs-folder-path>` (these are commits from this branch and main at some point of time)

Reply:
Still unclear why there is an A at the start and if that's unique to change files

Could the regex be something like
`/change(-beta)?\/@azure-communication-react-[a-f0-9\-]+\.json/gm` instead?

Otherwise add a comment, matching A in theory could match any line? I'm still v. unclear why there's a `A` at the start of those lines

Reply:
`Could the regex be something like
/change(-beta)?\/@azure-communication-react-[a-f0-9\-]+\.json/gm instead?`
I don't think that we need to test regex here as we always request the list of new files (`--diff-filter=A`) in the change log folders (path information).
The filtering was only for the empty lines that are part of the returned output from `git diff` command.
Basically the output looks like
```
Lines: [
  'A\tchange/@azure-communication-react-c7a5e0c3-2c59-4093-8334-9e705a8feb89.json',
  ''
]
Matches: [
  'A\tchange/@azure-communication-react-c7a5e0c3-2c59-4093-8334-9e705a8feb89.json'
]
Return (after `matches.map(line => line.split('\t')[1]);`)
 [
  'change/@azure-communication-react-c7a5e0c3-2c59-4093-8334-9e705a8feb89.json'
]
```

@JamesBurnside  I've update the code to filter out the empty lines and added comments. Please let me know if you still prefer the regex check.